pkg/router/template/router.go

```diff
@@ -222,18 +223,66 @@ func (r *templateRouter) EnableRateLimiter(interval int, handlerFunc ratelimiter
 	glog.V(2).Infof("Template router will coalesce reloads within %v seconds of each other", interval)
 }
 
-// writeDefaultCert is called a single time during init to write out the default certificate
+// secretToPem composes a PEM file at the output directory from an input private key and crt file.
+func secretToPem(outName string) error {
+	// The secret, when present, is mounted on /etc/pki/tls/private
+	// The secret has two components crt.tls and key.tls
+	// When the default cert is provided by the admin it is a pem
+	//   tls.crt is the supplied pem and tls.key is the key
+	//   extracted from the pem
+	// When the admin does not provide a default cert, the secret
+	//   is created via the service annotation. In this case
+	//   tls.crt is the cert and tls.key is the key
+	//   The crt and key are concatenated to form the needed pem
+
+	certDir := os.Getenv("DEFAULT_CERTIFICATE_DIR")
```

Comment:
This should be in pkg/cmd/infra/router/template.go.  It defines the command line args and sets how they bind into the go objects.